packages/docprovider/src/tokens.ts

```diff
@@ -45,3 +51,27 @@ export interface ISharedModelFactory extends Contents.ISharedFactory {
     factory: SharedDocumentFactory
   ): void;
 }
+
+/**
+ * The awareness interface.
+ *
+ * TODO: Move to @jupyter/YDoc
+ */
+export type IAwareness = Awareness;
+
+/**
+ * A provider interface for global awareness features.
+ */
+export interface IAwarenessProvider {
+  /**
+   * A signal to subscribe for incoming messages.
+   */
+  get chatMessage(): ISignal<this, IChatMessage>;
```

Comment:
Should we use the new stream instead of the signal?

Ref: https://lumino.readthedocs.io/en/latest/api/interfaces/signaling.IStream.html

Reply:
Should it be rather than a getter?

```suggestion
  readonly chatMessage: ISignal<this, IChatMessage>;
```

> This is the usual pattern in Lab

Reply:
I would also go for another naming like `messageStream`; something closer to what it provides. `chatMessage` could be misinterpreted, is it the latest message.